apps/api-extractor/src/api/IConfigFile.ts

```diff
@@ -0,0 +1,365 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+/**
+ * Determines how the TypeScript compiler engine will be invoked by API Extractor.
+ *
+ * @remarks
+ * This is part of the {@link IConfigFile} structure.
+ *
+ * @public
+ */
+export interface IConfigCompiler {
+
+  /**
+   * The root folder for the project.  This folder typically contains the tsconfig.json and package.json
+   * config files.
+   *
+   * @remarks
+   *
+   * The `rootFolder` path is resolved relative to the folder containing api-extractor.json.
+   *
+   * The default value for `rootFolder` is the token `<lookup>`, which means the folder is determined by traversing
+   * parent folders, starting from the folder containing api-extractor.json, and stopping at the first folder
+   * that contains a tsconfig.json file.  If a tsconfig.json file cannot be found in this way, then an error
+   * will be reported.
+   */
+  rootFolder: string;
+
+  /**
+   * Provides already parsed tsconfig.json contents.
+   *
+   * @remarks
+   * The value must conform to the TypeScript tsconfig schema:
+   *
+   * http://json.schemastore.org/tsconfig
+   *
+   * If omitted, then the tsconfig.json file will instead be read from the rootFolder.
+   */
+  overrideTsconfig?: { };
+
+  /**
+   * This option causes the compiler to be invoked with the `--skipLibCheck` option.
+   *
+   * @remarks
+   * This option is not recommended and may cause API Extractor to produce incomplete or incorrect declarations,
+   * but it may be required when dependencies contain declarations that are incompatible with the TypeScript engine
+   * that API Extractor uses for its analysis.  Where possible, the underlying issue should be fixed rather than
+   * relying on skipLibCheck.
+   */
+  skipLibCheck?: boolean;
+}
+
+/**
+ * Configures how the API review files (*.api.md) will be generated.
+ *
+ * @remarks
+ * This is part of the {@link IConfigFile} structure.
+ *
+ * @public
+ */
+export interface IConfigApiReport {
+  /**
+   * Whether to generate an API report.
+   */
+  enabled: boolean;
+
+  /**
+   * The filename for the API report files.  It will be combined with `reportFolder` or `reportTempFolder` to produce
+   * a full output filename.
+   *
+   * @remarks
+   * The file extension should be ".api.md", and the string should not contain a path separator such as `\` or `/`.
+   */
+  reportFileName?: string;
```

Comment:
Other config properties have the form `...FilePath` which combines the folder with the file name. Is there a reason these are split for API reports? Otherwise it might be more consistent to use a single `reportFilePath` property.

Reply:
People seem to have a lot of trouble understanding the workflow:

- **production**:  generate temp report file, compare to report file, fail if they are different
- **non-production**:  generate temp report file, compare to report file, if they are different then overwrite the report file with the temp file and issue a warning

The design idea was:  Enforce that the two files must have the same name.  One of them merely goes in a "temporary" folder.

My thinking was that this might make it more obvious that they really are same file.  Also it avoids a mistake where someone updates a `reportFilePath` setting but forgets to update the `reportTempFilePath`, and now you get an inconsistent message like this:

```
You have changed the public API signature for this project. 
Please overwrite etc/the-report.api.md with a copy of temp/your-library.api.md 
and then request an API review. See the Git repository README.md for more info.
```

Reply:
Ah ok, that makes a lot of sense.